tests/subsys/bluetooth/mesh/metadata_extraction/verify_metadata.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+import json
+import os
+import sys
+from zipfile import ZipFile
+
+
+class AnyInt:
+    def __eq__(self, other):
+        return isinstance(other, int)
+
+
+class WildcardString(str):
+    def __eq__(self, other):
+        return (isinstance(other, str) and len(self) == len(other) and
+                all([a in ('?', b) for a, b in zip(self, other)]))
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+EXPECTED_METADATA = [
+    {
+        'sign_version': {'major': 1, 'minor': 2, 'revision': 3, 'build_number': 4},
+        'binary_size': AnyInt(),
+        'composition_data': {
+            'cid': 1,
+            'pid': 2,
+            'vid': 3,
+            'crpl': 64,
+            'features': 5,
+            'elements': [
+                {'location': 1, 'sig_models': [0, 2], 'vendor_models': []}
+            ]
+        },
+        'composition_hash': '0x587a2fb',
+        'encoded_metadata': WildcardString('0102030004000000??????01fba287050100')
+    },
+    {
+        'sign_version': {'major': 1, 'minor': 2, 'revision': 3, 'build_number': 4},
+        'binary_size': AnyInt(),
+        'composition_data': {
+            'cid': 4,
+            'pid': 5,
+            'vid': 6,
+            'crpl': 64,
+            'features': 5,
+            'elements': [
+                {'location': 1, 'sig_models': [0, 2], 'vendor_models': []},
+                {'location': 2, 'sig_models': [4097, 4099], 'vendor_models': []}
+            ]
+        },
+        'composition_hash': '0x13f1a143',
+        'encoded_metadata': WildcardString('0102030004000000??????0143a1f1130200')
```

Comment:
I think it is better to extract the firmware size, put it here and then compare whole metadata.

Reply:
I thought about this, but the size extraction code would be literally identical to the code we're testing so I decided not to. Is there really any point to check that the same python code returns the same value?

Reply:
The code for encoding it would also be identical to the one we're testing

Reply:
But if the script change, we can catch it in the test